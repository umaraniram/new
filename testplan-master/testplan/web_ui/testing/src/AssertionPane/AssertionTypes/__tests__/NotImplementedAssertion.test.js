import React from 'react';
import {shallow} from 'enzyme';
import {StyleSheetTestUtils} from "aphrodite";

import NotImplementedAssertion from '../NotImplementedAssertion';

function defaultProps() {
  return {
    assertion: {
      "category": "DEFAULT",
      "machine_time": "2019-02-12T17:41:43.302500+00:00",
      "description": null,
      "line_no": 579,
      "meta_type": "entry",
      "type": "Unknown",
      "utc_time": "2019-02-12T17:41:43.302494+00:00"
    }
  };
}


describe('NotImplementedAssertion', () => {
  let props;
  let shallowComponent;

  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
    props = defaultProps();
    shallowComponent = undefined;
  });

  it('shallow renders the correct HTML structure', () => {
    shallowComponent = shallow(<NotImplementedAssertion {...props} />);
    expect(shallowComponent).toMatchSnapshot();
  });
});