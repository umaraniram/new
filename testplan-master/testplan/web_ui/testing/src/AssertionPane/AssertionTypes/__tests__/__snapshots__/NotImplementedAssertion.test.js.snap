// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotImplementedAssertion shallow renders the correct HTML structure 1`] = `
<Fragment>
  <FontAwesomeIcon
    border={false}
    className="icon_1tijhm4"
    fixedWidth={false}
    flip={null}
    icon="frown"
    inverse={false}
    key="faFrown"
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size="lg"
    spin={false}
    symbol={false}
    title=""
    transform={null}
  />
  Currently there is no rendering mechanism for this type of assertion. Please contact 
  <strong>
    the developers
  </strong>
   if you would like to have it implemented.
</Fragment>
`;
